node/bin/src/batcher/mod.rs

```diff
@@ -121,39 +121,26 @@ impl Batcher {
             self.pubdata_limit_bytes,
         );
 
-        // skip the blocks from already committed batches
-        // when the server restarts some historical blocks are replayed - they are already batched and committed
-        loop {
-            match self
-                .block_receiver
-                .peek_recv(|(_, replay_record, _)| {
-                    replay_record.block_context.block_number < self.first_block_to_process
-                })
-                .await
-            {
-                Some(true) => {
-                    // the block is before the first block to process, skip it
-                    let Some((_, replay_record, _)) = self.block_receiver.pop_buffer() else {
-                        anyhow::bail!("No block in buffer after peeking")
-                    };
-                    tracing::debug!(
-                        block_number = replay_record.block_context.block_number,
-                        "Skipping block before the first block to process",
-                    );
-                }
-                Some(false) => {
-                    // the block is within the range to process, proceed to batching
-                    break;
-                }
-                None => {
-                    anyhow::bail!("Batcher's block receiver channel closed unexpectedly");
-                }
+        match self
+            .block_receiver
+            .peek_recv(|(_, replay_record, _)| {
+                replay_record.block_context.block_number < self.first_block_to_process
```

Comment:
can we make the condition stronger and assert that `replay_record.block_context.block_number == self.first_block_to_process`?

In fact, we could have such `assert` on each incoming block - not only the first one. But we don't do it in every component - relying to the fact that they are properly ordered by upstream. 
That would mean having `let mut expected_block_number = self.first_block_to_process` and then asserting and bumping it on every block. 

On the first glance this may contradict my earlier suggestion that we want to skip "early" blocks **outside** of the main loop - but in this case we are not skipping but asserting that our assumption is correct - and it's worth doing it on every block not just startup.